src/cmd/singularity/actions_test.go

```diff
@@ -151,6 +151,10 @@ func testSTDINPipe(t *testing.T) {
 		{"sh", "permissions", []string{"-c", fmt.Sprintf("singularity exec %s id -u | grep `id -u`", imagePath)}, 0},
 		// testing run command properly hands arguments
 		{"sh", "arguments", []string{"-c", fmt.Sprintf("singularity run %s foo | grep foo", imagePath)}, 0},
+		// Stdin to URI based image
+		{"sh", "library", []string{"-c", "echo true | singularity shell library://busybox"}, 0},
+		{"sh", "docker", []string{"-c", "echo true | singularity shell docker://busybox"}, 0},
```

Comment:
busybox is just a single layer afaik?  Might be good to also test an action on a docker container that has multiple layers.  Maybe one of the ones that @dctrud setup that also has issues with whiteout files.

Reply:
We already test @dctrud whiteout issue at 

https://github.com/sylabs/singularity/blob/master/src/cmd/singularity/docker_test.go#L51

Reply:
fair enough.